pkg/pilot/cassandra/v3/pilot.go

```diff
@@ -67,12 +68,28 @@ func (p *Pilot) syncFunc(pilot *v1alpha1.Pilot) error {
 	return nil
 }
 
-func (p *Pilot) ReadinessCheck() error {
-	glog.V(2).Infof("readiness status: %q", "ok")
+func localNodeUpAndNormal(nodeTool nodetool.Interface) error {
+	nodes, err := nodeTool.Status()
+	if err != nil {
+		return err
+	}
+	localNode := nodes.LocalNode()
+	if localNode == nil {
+		return fmt.Errorf("Local node not found: %v", nodes)
+	}
+	if localNode.Status != nodetool.NodeStatusUp {
+		return fmt.Errorf("Unexpected local node status: %v", localNode.Status)
+	}
+	if localNode.State != nodetool.NodeStateNormal {
+		return fmt.Errorf("Unexpected local node state: %v", localNode.State)
+	}
 	return nil
 }
 
+func (p *Pilot) ReadinessCheck() error {
+	return localNodeUpAndNormal(p.nodeTool)
+}
+
 func (p *Pilot) LivenessCheck() error {
-	glog.V(2).Infof("liveness status: %q", "ok")
-	return nil
+	return localNodeUpAndNormal(p.nodeTool)
```

Comment:
The liveness check here probably shouldn't rely on the cassandra node being up and normal. If a node is recovering, or still starting up, the *readiness* probe should fail but liveness should pass. If a liveness probe fails, Kubernetes will begin restarting that pod, which can quite easily cause the pod to never start as a result of constant CrashLoopBackOff